packages/flutter/test/material/ink_sparkle_test.dart

```diff
@@ -0,0 +1,150 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This file is run as part of a reduced test set in CI on Mac and Windows
+// machines.
+@Tags(<String>['reduced-test-set'])
+
+import 'package:flutter/material.dart';
+import 'package:flutter/src/foundation/constants.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import '../rendering/mock_canvas.dart';
+
+void main() {
+  testWidgets('InkSparkle in a Button with default splashFactory paints by calling drawRect', (WidgetTester tester) async {
+    await tester.pumpWidget(MaterialApp(
+      home: Scaffold(
+        body: Center(
+          child: ElevatedButton(
+            style: ElevatedButton.styleFrom(splashFactory: InkSparkle.splashFactory),
+            child: const Text('Sparkle!'),
+            onPressed: () { },
+          ),
+        ),
+      ),
+    ));
+    final Finder buttonFinder = find.text('Sparkle!');
+    await tester.tap(buttonFinder);
+    await tester.pump();
+    await tester.pumpAndSettle();
```

Comment:
What is this test testing? (description says it's checking for drawRect, but the test doesn't seem to do that?)

Reply:
Added : expect(material, paintsExactlyCountTimes(#drawRect, 1));